---
title: 'MDI: Week 15 & 16'
author: Stephen Coleman
date: '2019-05-22'
slug: mdi-week-15-16
categories: []
tags: []
image:
  caption: ''
  focal_point: ''
bibliography: Weeks_15_16.bib
---



<p>This project is immensely enabled by the R programming language <span class="citation">(R Core Team 2019)</span>. Most data analysis and processing was carried out within R using a number of packages. The packages from <span class="citation">Wickham et al. (2019)</span> (data processing and wrangling), <span class="citation">Bache and Wickham (2014)</span> (data processing), <span class="citation">Wickham (2016)</span> (visualisation), <span class="citation">Müller and Wickham (2019)</span> (data storage) and <span class="citation">Dowle and Srinivasan (2019)</span> (reading, writing and processing data) deserve special mention as they tend to be imported in multiple of my scripts for multiple purposes.</p>
<p>I have fallen behind in tracking my progress, so starting afresh.</p>
<div id="mdi-tissue-specific-gene-sets" class="section level1">
<h1>MDI: Tissue specific gene sets</h1>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>We want to apply multiple dataset integration (MDI) <span class="citation">(P. Kirk et al. 2012)</span>, a Bayesian unsupervised clustering method, to gene expression data from various tissue / cell types to create tissue specific gene sets. We are using the data from the Correlated Expression and Disease Association Research (CEDAR) cohort <span class="citation">(The International IBD Genetics Consortium et al. 2018)</span>.</p>
<p>We have 9 .csv files, one for each tissue / cell type present of normalised gene expression data for 323 individuals. These are healthy individuals of European descent; the cohort consists of 182 women and 141 men with an average age of 56 years (but ranging from 19 to 86). None of the individuals are suffering from any autoimmune or inflammatory disease and were not taking corticosteroids or non-steroid anti-inflammatory drugs (with the exception of aspirin).</p>
<p>With regards to tissue types, samples from six circulating immune cells types:</p>
<ol style="list-style-type: decimal">
<li>CD4+ T lymphocytes (CD4);</li>
<li>CD8+ T lymphocytes (CD8);</li>
<li>CD14+ monocytes (CD14);</li>
<li>CD15+ granulocytes (CD15);</li>
<li>CD19+ B lymphocytes (CD19); and</li>
<li>platelets (PLA).</li>
</ol>
<p>Data from intestinal biopsies are also present, with sample taken from three distinct locations:</p>
<ol style="list-style-type: decimal">
<li>the illeum (IL);</li>
<li>the rectum (RE); and</li>
<li>the colon (TR).</li>
</ol>
<p>Not every individual is present in every dataset. However, as we are clustering genes this should not present a problem</p>
<p>Whole genome expression data were generated using HT-12 Expression Beadchips following the instructions of the manufacturer (Illumina). 29,464 autosomal probes (corresponding to 19,731 genes) were included across the datasets, but further thinning under various criteria reduced this further in each dataset. The fluorescence intensities were <span class="math inline">\(\log_2\)</span> transformed and Robust Spline Normalized with Lumi38.</p>
<p>It should be noted that some datasets are less information rich than others (for instance the platelets dataset has only around 8 thousand probes present).</p>
</div>
<div id="pipeline" class="section level2">
<h2>Pipeline</h2>
<p>Overview of steps currently implemented:</p>
<ol style="list-style-type: decimal">
<li><a href="https://github.com/stcolema/ra_chris_wallace/blob/master/Data_prep/download_data.R">Download</a> the <a href="http://139.165.108.18/srv/genmol/permanent/1be6993fe41c12a051c9244d67c91da2be49e5dd26a6cd79f442bc006971e2ef/CEDAR_GE/GE_Corr/crohn-index.html">data</a>;</li>
<li><a href="https://github.com/stcolema/ra_chris_wallace/blob/master/Data_prep/gene_subsetter_original_data.R">Subset the data</a> using the gene subsets (divided into big, med, small);</li>
<li><a href="https://github.com/stcolema/ra_chris_wallace/blob/master/Data_prep/matrix_transposer.R">Transpose the data and remove NAs</a>;</li>
<li>Inspect the data by <a href="https://github.com/stcolema/ra_chris_wallace/blob/master/Data_prep/pca_outlier_finding_gene_subsets.R">PCA</a> and remove outlier individuals for each dataset in each gene set;</li>
<li>To apply MDI we require that each dataset have the same row names in the same order, so we re-arrange our datasets to have common order of probes and include rows of 0’s for probes entirely missing using <a href="https://github.com/stcolema/ra_chris_wallace/blob/master/Data_prep/finad_all_probe_ids.R">this script</a>;</li>
<li>Apply MDI (either as implemented by <span class="citation">Mason et al. (2016)</span> or the original MATLAB code); and</li>
<li><a href="https://github.com/stcolema/ra_chris_wallace/blob/master/Analysis/analysis_script.R">Inspect the output</a> under several areas:
<ol style="list-style-type: decimal">
<li>Convergence of the <span class="math inline">\(\phi\)</span> parameter between datasets over iterations;</li>
<li>The disctribution of the <span class="math inline">\(\phi\)</span> parameter samples over MCMC iterations;</li>
<li>The posterior similarity matrix (PSM) for the clustering over MCMC iterations;</li>
<li>The adjusted rand index comparing the clustering in each iteration to the predicted clustering using <code>Simtocl()</code> from the <code>mcclust</code> <span class="citation">(Fritsch 2012)</span> package in R;</li>
<li>Inspect the clustering with the expression data using the <code>pheatmap</code> function <span class="citation">(Kolde 2019)</span>.</li>
</ol></li>
</ol>
</div>
<div id="transpose-the-data-and-remove-nas" class="section level2">
<h2>Transpose the data and remove NAs</h2>
<p>The data comes in the form [people <span class="math inline">\(\times\)</span> probes]. As we are interested in clustering genes (represented by the probes), we wish to transpose the data to fit the format required by MDI. Furthermore, MDI cannot handle NAs. We therefore wish to either remove NAs or impute a value in. As the missingness is not at random as if the expression levels are too low for a given reading it is left empty. This means we will fill NAs with the minimum expression value in the given row. However, if a probe is missing across a significant number of people (defined here as being above some fraction of the people present), we remove it from the dataset. We use an initial value of 0.1 for this threshold.</p>
<p>For the small and medium gene sets, no people or probes are removed under this criterion; for the big gene set we have the following loss:</p>
<table>
<thead>
<tr class="header">
<th></th>
<th align="center">Probes dropped</th>
<th align="center">People dropped</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CD14</td>
<td align="center">1</td>
<td align="center">0</td>
</tr>
<tr class="even">
<td>CD15</td>
<td align="center">2</td>
<td align="center">0</td>
</tr>
<tr class="odd">
<td>CD19</td>
<td align="center">3</td>
<td align="center">0</td>
</tr>
<tr class="even">
<td>CD4</td>
<td align="center">1</td>
<td align="center">0</td>
</tr>
<tr class="odd">
<td>CD8</td>
<td align="center">1</td>
<td align="center">0</td>
</tr>
<tr class="even">
<td>IL</td>
<td align="center">2</td>
<td align="center">0</td>
</tr>
<tr class="odd">
<td>PLA</td>
<td align="center">1</td>
<td align="center">0</td>
</tr>
<tr class="even">
<td>RE</td>
<td align="center">2</td>
<td align="center">0</td>
</tr>
<tr class="odd">
<td>TR</td>
<td align="center">1</td>
<td align="center">0</td>
</tr>
</tbody>
</table>
</div>
<div id="pca-and-visualisation" class="section level2">
<h2>PCA and visualisation</h2>
<p>We inspect the data to get some insight into underlying structure using PCA (as implemented in the <code>factoExtra</code> package by <span class="citation">Kassambara and Mundt (2017)</span>). This step can also be used to identify outliers that might be removed.</p>
</div>
<div id="re-arrange-the-order" class="section level2">
<h2>Re-arrange the order</h2>
<p>We want the same rownames and the order of rownames (here probe IDs) in each dataset. Some probes are not present in every dataset and these are added with an expression level of 0 (well below the minimum value representing observed data). We then put each dataset in the same order of probes.</p>
</div>
<div id="mdi" class="section level2">
<h2>MDI</h2>
<p>We have a choice of two flavours of MDI.</p>
<ol style="list-style-type: decimal">
<li>The C++ implementation from <span class="citation">Mason et al. (2016)</span>; or</li>
<li>The MATLAB implementation from <span class="citation">P. Kirk et al. (2012)</span> and used for glioblastoma subtyping by <span class="citation">Savage et al. (2013)</span>.</li>
</ol>
<p>The C++ version is faster and can handle the full 9 datasets. However, the MATLAB code includes a split-merge step <span class="citation">(Dahl 2005)</span> that is designed to help the program explore the full posterior space.</p>
<div id="c-implementation-and-many-seeds" class="section level3">
<h3>C++ implementation and many seeds</h3>
<p>We have observed that the C++ version is given to becoming stuck in a single clustering and not exploring the space much at all.</p>
<div class="figure">
<img src="/post/2019-03-04-mdi-week-4_files/iteration_heatplot_TR.png" alt="Clustering across iterations for each probe in the TR dataset. Notice how quickly it becomes stuck." />
<p class="caption">Clustering across iterations for each probe in the TR dataset. Notice how quickly it becomes stuck.</p>
</div>
<p>Ideally the MATLAB code should overcome this issue but we also consider model averaging. The idea is that we take <span class="math inline">\(n\)</span> start points for the C++ code, thin aggressively (on the basis that as the code becomes stuck quickly, the clustering in the final iteration is representative of the entire run) and treat this recorded clustering as a single draw from the posterior space and use the <span class="math inline">\(n\)</span> draws to decribe the clustering. To do this we use <span class="math inline">\(n\)</span> different random seeds and run for a small number of iterations (initially 500) and keep only the 500th iteration in each run. We then use <a href="https://github.com/stcolema/ra_chris_wallace/blob/master/Data_prep/big_csv.py">a python script</a> to collapse these values into one .csv file of the same format as if we had run <span class="math inline">\(n\)</span> iterations using the vanilla C++ implementation.</p>
</div>
<div id="matlab-implementation" class="section level3">
<h3>MATLAB implementation</h3>
<p>The MATLAB code uses more memory and thus is limited to at most 7 datasets. We remove PLA and CD15; PLA has a lot less gene expression data present (as one would expect in platelets as they lack a nucleus, so any RNA present is residual from before differentiation occurred), from previous analysis (see post 1 for details) we have seen that the CD15 dataset is less similar to the others and is thus unlikely to contribute much to an integrative clustering method.</p>
<div class="figure">
<img src="/post/weeks-15-16/2019-05-22-mdi-week-15-16_files/mdi_2_heatmap-1.png" alt="Heatmap of clustering for all 9 datasets and all probes. Notice how PLA has more empty probes (cluster 0 here) and that both CD15 and PLA are further away from the other datasets in the hierarchical clustering." />
<p class="caption">Heatmap of clustering for all 9 datasets and all probes. Notice how PLA has more empty probes (cluster 0 here) and that both CD15 and PLA are further away from the other datasets in the hierarchical clustering.</p>
</div>
<p>The MATLAB includes a split-merge step where at every <span class="math inline">\(i^{th}\)</span> iteration of MCMC, random clusters are attempted to be merged or split. This is meant to help the algorithm explore the posterior space more throoughly.</p>
</div>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<p>We inspect the PSMs as representative of the clustering including uncertainty. We inspect the <span class="math inline">\(\phi_{ij}\)</span> parameters from MDI; these are associated with two given datasets, dataset <span class="math inline">\(i\)</span> and dataset <span class="math inline">\(j\)</span> here, and are indivative of global similarity between them. <span class="math inline">\(\phi_{ij}\)</span> is a random variable sampled from a Gamma distribution where the shape parameter is defined by the number of observations (in our case probes) with the same label in the two datsets - thus the greater the <span class="math inline">\(\phi_{ij}\)</span> the more correlated the clustering between the datasets is.</p>
<p>We compare:</p>
<ol style="list-style-type: decimal">
<li>C++ MDI run over 5,000 different seeds and sampling the 500th iteration of each seed for 7 datasets;</li>
<li>MATLAB MDI for 1,225 iterations and samping every 25th iteration;</li>
<li>C++ MDI run over 3,000 different seeds and sampling the 1,000th iteration of each seed for 4 CD datasets (CD14, CD19, CD4 and CD8); and</li>
<li>MATLAB MDI run for 10,000 iterations and samping every 25th iteration for the 4 CD datasets.</li>
</ol>
<div id="phi-plotted-over-iterations" class="section level3">
<h3><span class="math inline">\(\phi\)</span> plotted over iterations</h3>
<p>We inspect a scatter plot of the values sampled for <span class="math inline">\(\phi\)</span> over iterations.</p>
<div id="datasets" class="section level4">
<h4>7 datasets</h4>
<div class="figure">
<img src="/post/weeks-15-16/Many_seeds_small_7_output_5000/Phi_series_plots/file_1_Phi_13.png" alt="\phi_{1,3} (between CD14 and CD4) after 500 iterations for 5,000 different initailsiations." />
<p class="caption"><span class="math inline">\(\phi_{1,3}\)</span> (between CD14 and CD4) after 500 iterations for 5,000 different initailsiations.</p>
</div>
<div class="figure">
<img src="/post/weeks-15-16/matlab_output_no_vsn_no_norm_n1225_t25/Phi_series_plots/file_1_Phi_13.png" alt="\phi_{1,3} (between CD14 and CD4) after 1,225 iterations for the MATLAB implementation." />
<p class="caption"><span class="math inline">\(\phi_{1,3}\)</span> (between CD14 and CD4) after 1,225 iterations for the MATLAB implementation.</p>
</div>
</div>
<div id="datasets-1" class="section level4">
<h4>4 datasets</h4>
<div class="figure">
<img src="/post/weeks-15-16/Many_seeds_4_cd/Phi_series_plots/file_1_Phi_13.png" alt="\phi_{1,3} (between CD14 and CD4) after 1,000 iterations for 3,000 different initailsiations and reduced to 4 datasets." />
<p class="caption"><span class="math inline">\(\phi_{1,3}\)</span> (between CD14 and CD4) after 1,000 iterations for 3,000 different initailsiations and reduced to 4 datasets.</p>
</div>
<div class="figure">
<img src="/post/weeks-15-16/Matlab_4_cd/Phi_series_plots/file_1_Phi_13.png" alt="\phi_{1,3} (between CD14 and CD4) after 10,000 iterations for the MATLAB implementation and reduced to 4 datasets." />
<p class="caption"><span class="math inline">\(\phi_{1,3}\)</span> (between CD14 and CD4) after 10,000 iterations for the MATLAB implementation and reduced to 4 datasets.</p>
</div>
</div>
</div>
<div id="distribution-of-phi-over-iterations" class="section level3">
<h3>Distribution of <span class="math inline">\(\phi\)</span> over iterations</h3>
<p>We inspect histograms of the sampled values for <span class="math inline">\(\phi\)</span>.</p>
<div id="datasets-2" class="section level4">
<h4>7 datasets</h4>
<div class="figure">
<img src="/post/weeks-15-16/Many_seeds_small_7_output_5000/Phi_histograms/Phi_13_histogram_plot.png" alt="Histogram of values of \phi_{1,3} (between CD14 and CD4) sampled in 500th iteration for 5,000 different initialisations." />
<p class="caption">Histogram of values of <span class="math inline">\(\phi_{1,3}\)</span> (between CD14 and CD4) sampled in 500th iteration for 5,000 different initialisations.</p>
</div>
<div class="figure">
<img src="/post/weeks-15-16/matlab_output_no_vsn_no_norm_n1225_t25/Phi_histograms/Phi_13_histogram_plot.png" alt="Histogram of values of \phi_{1,3} (between CD14 and CD4) sampled over 1,225 iterations with a thinning factor of 25 for the MATLAB implementation." />
<p class="caption">Histogram of values of <span class="math inline">\(\phi_{1,3}\)</span> (between CD14 and CD4) sampled over 1,225 iterations with a thinning factor of 25 for the MATLAB implementation.</p>
</div>
</div>
<div id="datasets-3" class="section level4">
<h4>4 datasets</h4>
<div class="figure">
<img src="/post/weeks-15-16/Many_seeds_4_cd/Phi_histograms/Phi_13_histogram_plot.png" alt="Histogram of values of \phi_{1,3} (between CD14 and CD4) sampled in 1,000th iteration for 3,000 different initialisations for 4 datasets." />
<p class="caption">Histogram of values of <span class="math inline">\(\phi_{1,3}\)</span> (between CD14 and CD4) sampled in 1,000th iteration for 3,000 different initialisations for 4 datasets.</p>
</div>
<div class="figure">
<img src="/post/weeks-15-16/Matlab_4_cd/Phi_histograms/Phi_13_histogram_plot.png" alt="Histogram of values of \phi_{1,3} (between CD14 and CD4) sampled over 10,000 iterations with a thinning factor of 25 for the MATLAB implementation applied to 4 datasets." />
<p class="caption">Histogram of values of <span class="math inline">\(\phi_{1,3}\)</span> (between CD14 and CD4) sampled over 10,000 iterations with a thinning factor of 25 for the MATLAB implementation applied to 4 datasets.</p>
</div>
</div>
</div>
<div id="adjusted-rand-index-plots" class="section level3">
<h3>Adjusted rand index plots</h3>
<p>Compare clustering in final iteration to all previous clusterings using the adjusted rand index.</p>
<div id="datasets-4" class="section level4">
<h4>7 datasets</h4>
<div class="figure">
<img src="/post/weeks-15-16/Many_seeds_small_7_output_5000/Adjusted_rand_index_plots/rand_index_plot_CD14.png" alt="Plot comparing the clustering in every recorded iteration / initialisation to the last recorded using the adjusted rand index for C++ MDI with 5,000 different initialisations over 7 datasets." />
<p class="caption">Plot comparing the clustering in every recorded iteration / initialisation to the last recorded using the adjusted rand index for C++ MDI with 5,000 different initialisations over 7 datasets.</p>
</div>
<div class="figure">
<img src="/post/weeks-15-16/matlab_output_no_vsn_no_norm_n1225_t25/Adjusted_rand_index_plots/rand_index_plot_CD14.png" alt="Plot comparing the clustering in every recorded iteration / initialisation to the last recorded using the adjusted rand index for MATLAB MDI with 1,225 iterations over 7 datasets." />
<p class="caption">Plot comparing the clustering in every recorded iteration / initialisation to the last recorded using the adjusted rand index for MATLAB MDI with 1,225 iterations over 7 datasets.</p>
</div>
</div>
<div id="datasets-5" class="section level4">
<h4>4 datasets</h4>
<div class="figure">
<img src="/post/weeks-15-16/Many_seeds_4_cd/Adjusted_rand_index_plots/rand_index_plot_CD14.png" alt="Plot comparing the clustering in every recorded iteration / initialisation to the last recorded using the adjusted rand index for C++ MDI with 3,000 different initialisations over 4 datasets." />
<p class="caption">Plot comparing the clustering in every recorded iteration / initialisation to the last recorded using the adjusted rand index for C++ MDI with 3,000 different initialisations over 4 datasets.</p>
</div>
<div class="figure">
<img src="/post/weeks-15-16/Matlab_4_cd/Adjusted_rand_index_plots/rand_index_plot_CD14.png" alt="Plot comparing the clustering in every recorded iteration / initialisation to the last recorded using the adjusted rand index for MATLAB MDI with 10,000 iterations over 4 datasets." />
<p class="caption">Plot comparing the clustering in every recorded iteration / initialisation to the last recorded using the adjusted rand index for MATLAB MDI with 10,000 iterations over 4 datasets.</p>
</div>
</div>
</div>
<div id="psms" class="section level3">
<h3>PSMs</h3>
<div id="datasets-6" class="section level4">
<h4>7 datasets</h4>
<div class="figure">
<img src="/post/weeks-15-16/Many_seeds_small_7_output_5000/Similarity_matrices/similarity_matrix_CD14.png" alt="Posterior similarity matrix for C++ MDI with 5,000 initialisations over 7 datasets." />
<p class="caption">Posterior similarity matrix for C++ MDI with 5,000 initialisations over 7 datasets.</p>
</div>
<div class="figure">
<img src="/post/weeks-15-16/matlab_output_no_vsn_no_norm_n1225_t25/Similarity_matrices/similarity_matrix_CD14.png" alt="Posterior similarity matrix for MATLAB MDI with 1,225 iterations over 7 datasets." />
<p class="caption">Posterior similarity matrix for MATLAB MDI with 1,225 iterations over 7 datasets.</p>
</div>
</div>
<div id="datasets-7" class="section level4">
<h4>4 datasets</h4>
<div class="figure">
<img src="/post/weeks-15-16/Many_seeds_4_cd/Similarity_matrices/similarity_matrix_CD14.png" alt="Posterior similarity matrix for C++ MDI with 3,000 initialisations over 4 datasets." />
<p class="caption">Posterior similarity matrix for C++ MDI with 3,000 initialisations over 4 datasets.</p>
</div>
<div class="figure">
<img src="/post/weeks-15-16/Matlab_4_cd/Similarity_matrices/similarity_matrix_CD14.png" alt="Posterior similarity matrix for MATLAB MDI with 10,000 iterations over 4 datasets." />
<p class="caption">Posterior similarity matrix for MATLAB MDI with 10,000 iterations over 4 datasets.</p>
</div>
</div>
</div>
<div id="gene-expression-data" class="section level3">
<h3>Gene expression data</h3>
<p>We include heatmaps of the gene expression data annotated by the clustering predicted by the PSMs. For the “Many seeds”&quot; approach there are too many clusters generated by <code>Simtocl()</code> to annotate - this is due to the uncertainty of the clustering.</p>
<div class="figure">
<img src="/post/weeks-15-16/Many_seeds_small_7_output_5000/Expression_heatmaps/pheatmap_all_datasets.png" />

</div>
<div class="figure">
<img src="/post/weeks-15-16/matlab_output_no_vsn_no_norm_n1225_t25/Expression_heatmaps/pheatmap_CD14.png" />

</div>
</div>
<div id="fused-genes" class="section level3">
<h3>Fused genes</h3>
<div id="datasets-8" class="section level4">
<h4>7 datasets</h4>
<p>For the many seeds implementation we have no fused genes between any datasets.</p>
<!-- ![](/post/weeks-15-16/Many_seeds_small_7_output_5000/Fusion_expression_data/heatmap_fused_genes_CD14_CD4.png) -->
<div class="figure">
<img src="/post/weeks-15-16/Many_seeds_small_7_output_5000/Fusion_expression_data/heatmap_unfused_genes_CD14_CD4.png" alt="Unfused genes between CD14 and CD4 for C++ MDI with 5,000 different initialisations and 7 datasets." />
<p class="caption">Unfused genes between CD14 and CD4 for C++ MDI with 5,000 different initialisations and 7 datasets.</p>
</div>
<div class="figure">
<img src="/post/weeks-15-16/matlab_output_no_vsn_no_norm_n1225_t25/Fusion_expression_data/heatmap_fused_genes_CD14_CD4.png" alt="Fused genes between CD14 and CD4 for MATLAB MDI with 1,225 iterations over 7 datasets." />
<p class="caption">Fused genes between CD14 and CD4 for MATLAB MDI with 1,225 iterations over 7 datasets.</p>
</div>
<div class="figure">
<img src="/post/weeks-15-16/matlab_output_no_vsn_no_norm_n1225_t25/Fusion_expression_data/heatmap_unfused_genes_CD14_CD4.png" alt="Unfused genes between CD14 and CD4 for MATLAB MDI with 1,225 iterations over 7 datasets." />
<p class="caption">Unfused genes between CD14 and CD4 for MATLAB MDI with 1,225 iterations over 7 datasets.</p>
</div>
</div>
<div id="datasets-9" class="section level4">
<h4>4 datasets</h4>
<div class="figure">
<img src="/post/weeks-15-16/Many_seeds_4_cd/Fusion_expression_data/heatmap_unfused_genes_CD14_CD4.png" alt="Unfused genes between CD14 and CD4 for C++ MDI with 3,000 different initialisations and 4 datasets." />
<p class="caption">Unfused genes between CD14 and CD4 for C++ MDI with 3,000 different initialisations and 4 datasets.</p>
</div>
<div class="figure">
<img src="/post/weeks-15-16/Matlab_4_cd/Fusion_expression_data/heatmap_fused_genes_CD14_CD4.png" alt="Fused genes between CD14 and CD4 for MATLAB MDI with 10,000 iterations over 4 datasets." />
<p class="caption">Fused genes between CD14 and CD4 for MATLAB MDI with 10,000 iterations over 4 datasets.</p>
</div>
<div class="figure">
<img src="/post/weeks-15-16/Matlab_4_cd/Fusion_expression_data/heatmap_unfused_genes_CD14_CD4.png" alt="Unfused genes between CD14 and CD4 for MATLAB MDI with 10,000 iterations over 4 datasets." />
<p class="caption">Unfused genes between CD14 and CD4 for MATLAB MDI with 10,000 iterations over 4 datasets.</p>
</div>
</div>
</div>
<div id="compare-clustering-in-matlab-to-mdi" class="section level3">
<h3>Compare clustering in MATLAB to MDI</h3>
<p>We compare the predicted clustering in the MATLAB implemenation to the clustering at each iteration of the Many seeds approach.</p>
<div class="figure">
<img src="/post/weeks-15-16/2019-05-22-mdi-week-15-16_files/rand_index_many_seeds_matlab.png" alt="Histograms of the distribution of the adjusted rand index between the clustering at each seed for C++ MDI compared to the predicted clustering of the MATLAB MDI, facetted by dataset." />
<p class="caption">Histograms of the distribution of the adjusted rand index between the clustering at each seed for C++ MDI compared to the predicted clustering of the MATLAB MDI, facetted by dataset.</p>
</div>
<div class="figure">
<img src="/post/weeks-15-16/2019-05-22-mdi-week-15-16_files/comparison_cpp_matlab_4_cd.png" alt="Histograms of the distribution of the adjusted rand index between the clustering at each seed for C++ MDI compared to the predicted clustering of the MATLAB MDI, facetted by dataset, for the 4 CD datasets." />
<p class="caption">Histograms of the distribution of the adjusted rand index between the clustering at each seed for C++ MDI compared to the predicted clustering of the MATLAB MDI, facetted by dataset, for the 4 CD datasets.</p>
</div>
<p>We compare the PSM for MATLAB and C++ (in the same ordering) and for the gene expression data. It appears that the clustering present in the C++ with many seeds is an uncertain version of the MATLAB. The MATLAB becomes stuck and has each member of each cluster allocated to the same cluster with each iteration (giving this binary similarity matrix). The many seeds PSM has many of the blocks / clusters visible in the MATLAB PSM, but their membership is less well defined - this method captures the uncertainty around the membership. This is reassuring - we can see that this model averaging does relate to the MATLAB while avoiding the problem of the spikey likelihood surface. However, this less distinct membership is harder to translate into clusters.</p>
<p>A small sense check is that the empty probes are all confidently assigned to the same cluster regardless of initialisation and method (the well defined block in the centre of the two PSMs).</p>
<p><img src="/post/weeks-15-16/2019-05-22-mdi-week-15-16_files/figure-html/comparison_mdi-1.png" width="672" /></p>
<p>We see that the gene expression for the CD14 data within the small gene set aligns nicely with the clusters. We see that there is more of a continuum between clusters for the many seeds approach - we notice that the top row of gene cluster most strongly with each other, but then with their neighbours in the expression space. Sometimes some of these genes cluster with the genes at the very bottom of the gene expression heatmap or with the band directly above the band of genes with an expression level of 0 across all people. This is satisfying - there is a relatively strong cluster (that one may pick out by eye in the expression data), but that the boundary between this and it’s neighbours is not overly distinct.</p>
<p><img src="/post/weeks-15-16/2019-05-22-mdi-week-15-16_files/figure-html/many_seed_psm_gene_expr-1.png" width="672" /></p>
</div>
</div>
<div id="bibliography" class="section level2 unnumbered">
<h2>Bibliography</h2>
<div id="refs" class="references">
<div id="ref-bache2014magrittr">
<p>Bache, Stefan Milton, and Hadley Wickham. 2014. <em>Magrittr: A Forward-Pipe Operator for R</em>. <a href="https://CRAN.R-project.org/package=magrittr" class="uri">https://CRAN.R-project.org/package=magrittr</a>.</p>
</div>
<div id="ref-dahl_sequentially-allocated_2005">
<p>Dahl, David B. 2005. “Sequentially-Allocated Merge-Split Sampler for Conjugate and Nonconjugate Dirichlet Process Mixture Models,” 27.</p>
</div>
<div id="ref-dowle2019data">
<p>Dowle, Matt, and Arun Srinivasan. 2019. <em>Data.table: Extension of ‘Data.frame‘</em>. <a href="https://CRAN.R-project.org/package=data.table" class="uri">https://CRAN.R-project.org/package=data.table</a>.</p>
</div>
<div id="ref-fritsch2012mcclust">
<p>Fritsch, Arno. 2012. <em>Mcclust: Process an Mcmc Sample of Clusterings</em>. <a href="https://CRAN.R-project.org/package=mcclust" class="uri">https://CRAN.R-project.org/package=mcclust</a>.</p>
</div>
<div id="ref-kassambara2017factoextra">
<p>Kassambara, Alboukadel, and Fabian Mundt. 2017. <em>Factoextra: Extract and Visualize the Results of Multivariate Data Analyses</em>. <a href="http://www.sthda.com/english/rpkgs/factoextra" class="uri">http://www.sthda.com/english/rpkgs/factoextra</a>.</p>
</div>
<div id="ref-kirk_bayesian_2012">
<p>Kirk, Paul, Jim E. Griffin, Richard S. Savage, Zoubin Ghahramani, and David L. Wild. 2012. “Bayesian Correlated Clustering to Integrate Multiple Datasets.” <em>Bioinformatics</em> 28 (24): 3290–7. doi:<a href="https://doi.org/10.1093/bioinformatics/bts595">10.1093/bioinformatics/bts595</a>.</p>
</div>
<div id="ref-kolde2019pheatmap">
<p>Kolde, Raivo. 2019. <em>Pheatmap: Pretty Heatmaps</em>. <a href="https://CRAN.R-project.org/package=pheatmap" class="uri">https://CRAN.R-project.org/package=pheatmap</a>.</p>
</div>
<div id="ref-mason2016mdi">
<p>Mason, Samuel A, Faiz Sayyid, Paul DW Kirk, Colin Starr, and David L Wild. 2016. “MDI-Gpu: Accelerating Integrative Modelling for Genomic-Scale Data Using Gp-Gpu Computing.” <em>Statistical Applications in Genetics and Molecular Biology</em> 15 (1). De Gruyter: 83–86.</p>
</div>
<div id="ref-muller2019tibble">
<p>Müller, Kirill, and Hadley Wickham. 2019. <em>Tibble: Simple Data Frames</em>. <a href="https://CRAN.R-project.org/package=tibble" class="uri">https://CRAN.R-project.org/package=tibble</a>.</p>
</div>
<div id="ref-R2019">
<p>R Core Team. 2019. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>.</p>
</div>
<div id="ref-savage_identifying_2013">
<p>Savage, Richard S., Zoubin Ghahramani, Jim E. Griffin, Paul Kirk, and David L. Wild. 2013. “Identifying Cancer Subtypes in Glioblastoma by Combining Genomic, Transcriptomic and Epigenomic Data.” <em>arXiv:1304.3577 [Q-Bio, Stat]</em>, April. <a href="http://arxiv.org/abs/1304.3577" class="uri">http://arxiv.org/abs/1304.3577</a>.</p>
</div>
<div id="ref-the_international_ibd_genetics_consortium_ibd_2018">
<p>The International IBD Genetics Consortium, Yukihide Momozawa, Julia Dmitrieva, Emilie Théâtre, Valérie Deffontaine, Souad Rahmouni, Benoît Charloteaux, et al. 2018. “IBD Risk Loci Are Enriched in Multigenic Regulatory Modules Encompassing Putative Causative Genes.” <em>Nature Communications</em> 9 (1): 2427. doi:<a href="https://doi.org/10.1038/s41467-018-04365-8">10.1038/s41467-018-04365-8</a>.</p>
</div>
<div id="ref-wickham2016ggplot2">
<p>Wickham, Hadley. 2016. <em>Ggplot2: Elegant Graphics for Data Analysis</em>. Springer-Verlag New York. <a href="https://ggplot2.tidyverse.org" class="uri">https://ggplot2.tidyverse.org</a>.</p>
</div>
<div id="ref-wickham2019dplyr">
<p>Wickham, Hadley, Romain François, Lionel Henry, and Kirill Müller. 2019. <em>Dplyr: A Grammar of Data Manipulation</em>. <a href="https://CRAN.R-project.org/package=dplyr" class="uri">https://CRAN.R-project.org/package=dplyr</a>.</p>
</div>
</div>
</div>
</div>
